### ISSUE TITLE
Crashing on Wayland

### ISSUE THREAD

**Issue by tuqqo** [DavidoTek/ProtonUp-Qt#312]
Please fill out following when reporting a new bug:

**Describe the bug**  
App is crashing when selecting something in any dropdown menu.

**To Reproduce**  
Start App, select any Engine (steam for example), click Add Version, select something in the dropdown
-> Crash

**Expected behavior**  
No crash

**Desktop (please complete the following information):**  
 - **Platform**: Chromebook
 - **System**: ChromeOS 119
 - **Version**: ProtonUp-Qt 2.8.2
 - **How did you install ProtonUp-Qt?**: Flatpak

**Terminal output**  
```
qt.pysideplugin: Environment variable PYSIDE_DESIGNER_PLUGINS is not set, bailing out.
qt.pysideplugin: No instance of QPyDesignerCustomWidgetCollection was found.
Gamepad error: No gamepad found.
qt.qpa.wayland: Ignoring unexpected wl_surface.enter received for output with id: 7 screen name: "Screen5" screen model: "unknown" This is most likely a bug in the compositor.
qt.pysideplugin: No instance of QPyDesignerCustomWidgetCollection was found.
qt.qpa.wayland: Ignoring unexpected wl_surface.enter received for output with id: 7 screen name: "Screen5" screen model: "unknown" This is most likely a bug in the compositor.
qt.qpa.wayland: Ignoring unexpected wl_surface.enter received for output with id: 7 screen name: "Screen5" screen model: "unknown" This is most likely a bug in the compositor.
The Wayland connection broke. Did the Wayland compositor die?
```

Propably a QT Problem ?!
https://github.com/eliranwong/UniqueBible/issues/68
https://bugs.launchpad.net/bugs/2043867

**Comment by sonic2kk** [comment]
Likely related to Qt, ProtonUp-Qt uses PySide6 and doesn't talk to anything as low level as Wayland. Also, I'm not sure if ProtonUp-Qt "officially" targets ChromeOS, so that could be an issue.

I'm not sure that this issue is specific to ProtonUp-Qt or something it can fix, it may be fixed by a PySide / KDE Runtime version bump, but that's really an upstream issue.

**Comment by tuqqo** [comment]
Yea i kinda thought the same. It's not the first ChromeOS upstream problem that magically fixes itself after some random update 🤷🏻‍♂️ thanks for the workaround tho!! Gonna try later 👍

**Comment by DavidoTek** [comment]
> @DavidoTek We should keep this in mind before merging https://github.com/flathub/net.davidotek.pupgui2/pull/24, I'm not sure that the fallback handles crashes like this, my guess is that it would not

Yeah, that is the reason why I'm still stalling. Let's wait for @tuqqo 's response what happens when disabling Wayland.

> As a workaround, you can disable the Wayland socket in a tool such as Flatseal

Also @tuggo, can you try to disable `x11` and instead enable both `wayland` and `fallback-x11` and see what happens?

> I'm not sure if ProtonUp-Qt "officially" targets ChromeOS

I haven't actively added features for it, but as long it doesn't break anything else, I guess we can at least maintain it in a working state. If there is anything else to consider/improve, I'm open for it.
I wonder, does ChromeOS support Wayland at all, and is the Linux support still actively supported? Afaik it is some sort of Ubuntu running in a Docker/chroot, right?

**Comment by sonic2kk** [comment]
I was unsure if ChromeOS supported Wayland, I just took OPs word for it, though it surprised me I also could see that being the case depending on how the Linux support is setup.

I think the Linux side is still supported, as I still see people on ChromeOS trying to run games via Proton on the Proton issue tracker, making reference to the Mesa drivers as well. I'm not too sure how it's set up, but the way you've described it does sound about right.

It seems like Wayland was actually used for Android and Linux apps, but actually, it looks like [ChromeOS may soon use Wayland by default everywhere](https://arstechnica.com/gadgets/2023/08/google-is-finally-separating-chrome-from-chromeos-for-easier-updates/).

> Previously ChromeOS was using a homemade graphics stack called "[Freon](https://iwf1.com/chrome-os-implements-a-new-home-made-graphics-stack-named-freon-ditches-x11)," but now with Wayland, it'll be on the new and normal desktop Linux graphics stack. [...] Google says this represents "more Wayland support" because Wayland was previously used for Android and Linux apps, but now it'll be used for the native Chrome OS graphics, too.

I don't own a Chomebook to verify but it seems like Linux applications do use Wayland on ChromeOS. That doesn't mean they have support for Xwayland though, and however their Wayland support is implemented may either be buggy, or may be exposing upstream bugs.

**Comment by tuqqo** [comment]
Hey, sorry I'm not able to test until tomorrow at my Chromebook. But yes, ChromeOS is powered by Wayland. But it's not yet finished as mesa drivers are still an issue cause of Vulkan API. Once finished it will be full Vulkan API on Wayland. And yes, until then there are always a couple of upstream problems like this, when one thing depends on something else. Basically 95% of the Linux container runs perfectly. And no it's not docker afaik, they are past that already.

**Comment by tuqqo** [comment]
ok, im done testing.

wayland off + x11 on = working
x11 off + wayland on + xwayland on = crash

**Comment by tuqqo** [comment]
uname -a
Linux penguin 6.1.55-06876-gecce3ad0352d #1 SMP PREEMPT_DYNAMIC Sat Nov 11 18:44:44 PST 2023 x86_64 GNU/Linux

cat /proc/version
Linux version 6.1.55-06876-gecce3ad0352d (chrome-bot@chromeos-release-builder-us-central1-b-x32-23-y0bg) (Chromium OS 17.0_pre498229-r6 clang version 17.0.0 (/mnt/host/source/src/third_party/llvm-project 14f0776550b5a49e1c42f49a00213f7f3fa047bf), LLD 17.0.0) #1 SMP PREEMPT_DYNAMIC Sat Nov 11 18:44:44 PST 2023

env | grep -E -i 'x11|xorg|wayland' 
WAYLAND_DISPLAY_LOW_DENSITY=wayland-1
WAYLAND_DISPLAY=wayland-0
loqee@penguin:~$

**Comment by DavidoTek** [comment]
> It seems like Wayland was actually used for Android and Linux apps, but actually, it looks like ChromeOS may soon use Wayland by default everywhere.
> the debian container on ChromeOS (crostini) is powered by Wayland

That's good I guess.

> I believe this application also uses Qt, either with PySide6 if available, and falling back to pyqt otherwise (repo search link)
> In their commit that supposedly fixes the Wayland compositor crash, I cannot discern what exactly fixes it

Hmm interesting. We should keep that in mind then.
I don't feel like implementing two different Main Windows is a good idea. We should look into what is actually making the difference and see if it possible to implement than in a util function or something.

> Main ChromeOS (crouton, based on gentoo) with it's debian Linux Subsystem (crostini) and the arcvm android container.

Ah okay.

> x11 on + wayland off + xwayland fallback off = working
> x11 off + wayland off + xwayland fallback on = working

Seems like it is only working when Wayland is disabled and it is using X11.

**Comment by sonic2kk** [comment]
Yeah, I definitely don't think two main windows is a maintainable solution either.

It's good to know that it works without X11 and that for now this is a viable workaround on ChromeOS. I don't think we can really handle telling Qt which to pick unfortunately. Since ChromeOS runs things in containers, setting an environment variable really early on if we "detect" ChromeOS would be very flimsy as well.

We'd have to find out what's causing this issue on ChromeOS, and beyond that, we'd also be best removing it once this is fixed. I'm not sure if it's Qt or ChromeOS' Wayland compositor causing the crash - it could just be a poorly implemented compositor, or Qt could be making some incorrect assumptions about whet the compositor implements. It's not uncommon for applications to require a certain Wayland version either which ChromeOS may not meet. For example, [GameScope requires Wayland Server >= 1.21, and Wayland Protocols >= v1.17](https://github.com/ValveSoftware/gamescope/blob/cf19852ce333fb2a23cc18077b63ed55578d9839/src/meson.build#L14). I couldn't find if Qt6.5 which I believe our PySide6 version uses, has any such minimum Wayland version requirements. Likewise, I'm not sure what version of Wayland that ChromeOS' container(s?) ship with.

There could be a number of issues here, and what I wanted to highlight here is that some of it may be out of an application maintainer's hands to fix, and also that even if we can fix it, the "real" fix will hopefully, eventually, come from upstream, so ideally ChromeOS won't become a maintenance burden and either ChromeOS or Qt will support Wayland on the desktop in such a way that this is properly abstracted away.

**Comment by tuqqo** [comment]
> There could be a number of issues here, and what I wanted to highlight here is that some of it may be out of an application maintainer's hands to fix, and also that even if we can fix it, the "real" fix will hopefully, eventually, come from upstream, so ideally ChromeOS won't become a maintenance burden and either ChromeOS or Qt will support Wayland on the desktop in such a way that this is properly abstracted away.

Most likely, yes. It's great that theres a workaround and you replied so fast, do I could continue with steam tinker launcher 🙏

And yea, until Crostini got it's true dedicated vulkan based mesa driver, it's probably a waste of time fiddling with upstream errors.

**Comment by sonic2kk** [comment]
Google are [discontinuing support for Steam on ChromeOS](https://9to5google.com/2025/08/07/steam-chromebook-2026/) in 2026 (just a few months away, spooky!), so is this issue still relevant? I wonder if we can go ahead with x11-fallback then?
